src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRangeTranslator.cs

```diff
@@ -0,0 +1,110 @@
+            TryTranslateOperator(methodCallExpression);
+
+        /// <summary>
+        /// Attempts to translate the <see cref="MethodCallExpression"/> as a PostgreSQL range operator.
+        /// </summary>
+        /// <param name="expression">The <see cref="MethodCallExpression"/> to be translated.</param>
+        /// <returns>
+        /// The expression if successful; otherwise, null.
+        /// </returns>
+        [CanBeNull]
+        static Expression TryTranslateOperator([NotNull] MethodCallExpression expression)
+        {
+            switch (expression.Method.Name)
+            {
+            case nameof(NpgsqlRangeExtensions.Contains):
+                return MakeBinaryExpression(expression, "@>", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.ContainedBy):
+                return MakeBinaryExpression(expression, "<@", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.Overlaps):
+                return MakeBinaryExpression(expression, "&&", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.IsStrictlyLeftOf):
+                return MakeBinaryExpression(expression, "<<", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.IsStrictlyRightOf):
+                return MakeBinaryExpression(expression, ">>", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.DoesNotExtendRightOf):
+                return MakeBinaryExpression(expression, "&<", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.DoesNotExtendLeftOf):
+                return MakeBinaryExpression(expression, "&>", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.IsAdjacentTo):
+                return MakeBinaryExpression(expression, "-|-", typeof(bool));
+
+            case nameof(NpgsqlRangeExtensions.Union):
+                return MakeBinaryExpression(expression, "+", expression.Arguments[0].Type);
+
+            case nameof(NpgsqlRangeExtensions.Intersect):
+                return MakeBinaryExpression(expression, "*", expression.Arguments[0].Type);
+
+            case nameof(NpgsqlRangeExtensions.Except):
+                return MakeBinaryExpression(expression, "-", expression.Arguments[0].Type);
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a <see cref="CustomBinaryExpression"/>.
+        /// </summary>
+        /// <param name="expression">The <see cref="MethodCallExpression"/> containing two parameters.</param>
+        /// <param name="symbol">The symbolic operator for PostgreSQL.</param>
+        /// <param name="returnType">The return type of the operator.</param>
+        /// <returns>
+        /// A <see cref="CustomBinaryExpression"/>.
+        /// </returns>
+        [NotNull]
+        static Expression MakeBinaryExpression([NotNull] MethodCallExpression expression, [NotNull] string symbol, [NotNull] Type returnType) =>
```

Comment:
Similar to the above, why not just instantiate `CustomBinaryExpression` inside the switch, rather than calling this method which does it? There's almost zero value added here.